pymatgen/analysis/Fstar/fstar.py

```diff
@@ -0,0 +1,246 @@
+"""
+This module implements an f* coordinate and diagram generator.
+For information about f* diagrams, see the following publications:
+Yin, L. et al. Extending the limits of powder diffraction analysis: diffraction parameter space, occupancy defects, and
+atomic form factors. Rev. Sci. Instrum. 89, 093002 (2018). 10.1063/1.5044555
+Yin, L. et al.Thermodynamics of Antisite Defects in Layered NMC Cathodes: Systematic Insights from High-Precision Powder
+Diffraction Analyses Chem. Mater 2020 32 (3), 1002-1010. 10.1021/acs.chemmater.9b03646
+
+"""
+
+import os
+import numpy as np
+import pandas as pd
+from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+from pymatgen.io.cif import CifParser, str2float
+import plotly.express as px
+
+
+# Load in the form factors
+
+with open(os.path.join(os.path.dirname(__file__),
+                       "xray_factors_2.csv")) as f:
+    X_RAY_SCATTER_DF = pd.read_csv(f)
+    # from https://it.iucr.org/Cb/ch6o1v0001/ table 6.1.1.4
+with open(os.path.join(os.path.dirname(__file__),
+                       "neutron_factors.csv")) as f:
+    NEUTRON_SCATTER_DF = pd.read_csv(f)
+    # from http://www.ccp14.ac.uk/ccp/web-mirrors/neutrons/n-scatter/n-lengths/LIST~1.HTM
+
+
+class FStarDiagram:
+    """
+    Take a list of structure objects and/or cifs and use them to generate an f* phase diagram.
+    """
+
+    def __init__(self, structure_objects=None, cifs=None, cif_index=0, occupancy_tolerance=5,

```

Comment:
I'm reviewing this again before merging, I think this `__init__` should be simplified to be just:

```
def __init__(self, structures, scattering_type='X-ray_simple', custom_scatter=None)
```

If the structures come from a CIF, when a person is using this class they would write something like:

```
structures = [CifParser(file, occupancy_tolerance=5).get_structures(primitive=False)[0] for file in cifs]
fstar = FStarDiagram(structures)
```

I know this is an extra line for the end-user, but I get the feeling that the CIF input (e.g. especially cif-index etc) is very specific to where your CIFs are coming from.

The philosophy is that the `Structure` is the canonical way that pymatgen deals with structural information internally.